controllers/mirroring/mirroring_controller.go

```diff
@@ -0,0 +1,478 @@
+	}
+
+	disableRBDMirror := disableMirroring || len(storageConsumers.Items) >= 1
+
+	if !disableRBDMirror {
+		_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, rbdMirror, func() error {
+			if err := r.own(clientMappingConfig, rbdMirror); err != nil {
+				return err
+			}
+			rbdMirror.Spec.Count = 1
+			return nil
+		})
+		if err != nil {
+			r.log.Error(err, "Failed to create/update the CephRBDMirror", "CephRBDMirror", rbdMirror)
+			return err
+		}
+	} else {
+		if err := r.delete(rbdMirror); err != nil {
+			return fmt.Errorf("failed to delete CephRBDMirror: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func (r *MirroringReconciler) reconcileBlockPoolMirroring(
+	ocsClient *providerClient.OCSProviderClient,
+	clientMappingConfig *corev1.ConfigMap,
+	storageClusterPeer *ocsv1.StorageClusterPeer,
+	disableMirroring bool,
+) (ctrl.Result, error) {
+
+	cephBlockPoolsList := &rookCephv1.CephBlockPoolList{}
+	if err := r.list(
+		cephBlockPoolsList,
+		client.InNamespace(clientMappingConfig.Namespace),
+	); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to list cephBlockPools: %v", err)
+	}
+
+	blockPoolByName := map[string]*rookCephv1.CephBlockPool{}
+	for i := range cephBlockPoolsList.Items {
+		cephBlockPool := &cephBlockPoolsList.Items[i]
+		if _, ok := cephBlockPool.GetLabels()[util.ForbidMirroringLabel]; !ok {
+			continue
+		}
+		blockPoolByName[cephBlockPoolsList.Items[i].Name] = &cephBlockPoolsList.Items[i]
+	}
+
+	// fetch BlockPoolsInfo
+	response, err := ocsClient.GetBlockPoolsInfo(
+		r.ctx,
+		storageClusterPeer.Status.PeerInfo.StorageClusterUid,
+		maps.Keys(blockPoolByName),
+	)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to get CephBlockPool(s) info from Peer: %w", err)
+	}
+
+	for i := range response.BlockPoolsInfo {
+		blockPoolName := response.BlockPoolsInfo[i].BlockPoolName
+		cephBlockPool := blockPoolByName[blockPoolName]
+
+		mirroringToken := response.BlockPoolsInfo[i].MirroringToken
+		secretName := fmt.Sprintf("mirroring-token-%s", blockPoolName)
+		if mirroringToken != "" {
+			mirroringSecret := &corev1.Secret{}
+			mirroringSecret.Name = secretName
+			mirroringSecret.Namespace = clientMappingConfig.Namespace
+			var err error
+
+			_, err = ctrl.CreateOrUpdate(r.ctx, r.Client, mirroringSecret, func() error {
+				if err = r.own(clientMappingConfig, mirroringSecret); err != nil {
+					return err
+				}
+				mirroringSecret.Data = map[string][]byte{
+					"pool":  []byte(blockPoolName),
+					"token": []byte(mirroringToken),
+				}
+				return nil
+			})
+			if err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to create/update bootstrap secret: %w", err)
+			}
+		}
+
+		_, err = controllerutil.CreateOrUpdate(r.ctx, r.Client, cephBlockPool, func() error {
+			util.AddAnnotation(
+				cephBlockPool,
+				util.BlockPoolMirroringTargetIDAnnotation,
+				response.BlockPoolsInfo[i].BlockPoolID,
+			)
+
+			if disableMirroring {
+				cephBlockPool.Spec.Mirroring = rookCephv1.MirroringSpec{}
```

Comment:
I think this check needs to be at a higher scope as there is no reason to run all of the code up until this point if we are disabling mirroring